### ISSUE TITLE
Chip does't selects the object from dropdown upon pressing enter (v0.10.1)

### ISSUE THREAD

**Issue by navneet35371** [angular/material#4404]
![screenshot-2](https://cloud.githubusercontent.com/assets/4524905/9602579/7efa1838-50c7-11e5-9f88-ac5643f749eb.png)

and then when i press enter, it doesn't show up in the selected list. whereas mouse click works fine.

![screenshot-3](https://cloud.githubusercontent.com/assets/4524905/9602580/7efb9b72-50c7-11e5-8b70-b0e222e81719.png)

**Comment by langdonx** [comment]
I was just dealing with this same issue.  It looks like `MdChipsCtrl.prototype.inputKeydown` is conflicting with `MdAutocompleteCtrl.keydown`.

**Comment by topherfangio** [comment]
This should be fixed in the latest master branch (although there is another separate issue with the dropdown not showing any contents: #4390).

Can you double-check that it is working if you use the latest code? If not, can you please tell us what browser you are using?

**Comment by langdonx** [comment]
What's in master partially fixes the issue.  It works as expected when I press `TAB`, but when I press `ENTER`, I get an empty chip along with my desired chip.  It's similar to what happens on the demo site right now where if you type "q" (a letter not found in anyone's name) and press enter.

I fixed the issue by checking `event.defaultPrevented` in the first case of `MdChipsCtrl.prototype.inputKeydown`.

    if (event.defaultPrevented) return; // you guys should always use braces =P

**Comment by topherfangio** [comment]
@langdonx It's not actually missing the email; it's just being cut off due to some CSS/sizing issues I haven't had a chance to look into yet.

Regarding the tab/enter issue: what browser are you using? I'm not seeing this in Chrome.

**Comment by langdonx** [comment]
[On the demo site](https://material.angularjs.org/latest/#/demo/material.components.chips), in the contact chips, you can type: `q enter w enter e enter r enter t enter y enter` etc to get 6+ blank chips.

This happens in Chrome 44, Canary, Firefox 40, and IE11.

After pulling down master and using the same code from the demo site, instead of 6 blank chips, you end up with: (blank) (blank) (Some-guy...) (blank) (blank) (blank) (blank) (Narayana Garner)

If I type refresh and just type `o enter`, I end up with (blank) (Oddr Sarno).

If I change the demo code from `ng-require-match` to `ng-require-match="true"`, it stops doing this.

I imagine ngRequireMatch isn't meant to be used as a value-less attribute at this point.

**Comment by topherfangio** [comment]
@langdonx It looks like the demo was updated a while back in https://github.com/angular/material/commit/86db4db9fd6e61314c1ce08cbffdffd52eae787f to fix the `ng-require-match`.

Can you please test one more time against the master version of the demo available at https://material.angularjs.org/HEAD/#/demo/material.components.chips to see if it is working as expected (aside from the known bug of the dropbown items still being blank)?

**Comment by langdonx** [comment]
@topherfangio The original issue here has been resolved, so you can probably close this issue.

The `defaultPrevented` issue is still outstanding (mentioned previously here and referenced in #4412).  Using the Edit on CodePen link and removing `ngRequireMatch`, two chips get added when choosing an existing contact, and a blank chip gets added when no contact exists with your search term.  I'm not really sure contact chips are meant to be used without `ngRequireMatch` though -- seems like an odd workflow.

**Comment by navneet35371** [comment]
@topherfangio Yeah.. the original issue seems to be resolved!! aside from dropdown items still being blank. This issue can be closed. :100:

**Comment by bopm** [comment]
Even while it fixed, whole release process in Angular Material is broken. So now 0.10.1 have that issue, while HEAD have it fixed but instead of that provide us with selectors rework that breaks multiple things at the same time. How about start to provide us with fixes that have nothing to do with new features in older versions? This issue seems pretty good for 0.10.2, not only 0.11.0.

**Comment by ThomasBurleson** [comment]
@bopm - Are you asking for selected fixes to be applied to previous releases? If yes, we cannot accommodate that request: our team is rapidly moving forward for the v1.0 release and we will not support previous releases.

**Comment by langdonx** [comment]
@bopm *Warning: This component is a WORK IN PROGRESS. If you use it now, it will probably break on you in the future.* via [material.angularjs.org](https://material.angularjs.org/latest/#/api/material.components.chips/directive/mdChips)

**Comment by bopm** [comment]
@ThomasBurleson I understand your position. But there is no stable release date for v1.0, things get pushed forward to next possible releases between now and 1.0 every now and then. That leads to a constant 'we need to wait till next release' changing with 'release is there but now all broken again in a different part of app'. If there is releases, not betas, please provide at least basic support for them.  

@langdonx it's more or less about possible API and appearance changes, not about 'this component will never work till v1.0, please avoid', or I get it all wrong?

TL;DR There is common release management practices. Delivering things as hotfixes one of them. Please separate hotfixes from things that rely on new features. This will make usage of project simpler and more comfortable.

**Comment by topherfangio** [comment]
Thanks for the feedback all; since we have other issues to track the other problems, I am closing this one :-)
